docs/investigation/multi-cluster/admiralty.md

```diff
@@ -0,0 +1,186 @@
+    IP=$(docker inspect eu-control-plane \
+      --format "{{ .NetworkSettings.Networks.kind.IPAddress }}")
+    
+    # iv. prepare a kubeconfig using the token and address found above, and the server certificate from your kubeconfig (luckily also valid for this address, not just the address in your kubeconfig),
+    CONFIG=$(kubectl --context kind-eu config view \
+      --minify --raw --output json | \
+      jq '.users[0].user={token:"'$TOKEN'"} | .clusters[0].cluster.server="https://'$IP':6443"')
+    
+    # v. save the prepared kubeconfig in a secret in the management cluster:
+    kubectl --context kind-dev-capact create secret generic eu \
+      --from-literal=config="$CONFIG"
+    ```
+
+5. In the Capact cluster, create a Target for workload cluster:
+
+    ```yaml
+    cat <<EOF | kubectl --context kind-dev-capact apply -f -
+    apiVersion: multicluster.admiralty.io/v1alpha1
+    kind: Target
+    metadata:
+      name: eu
+    spec:
+      kubeconfigSecret:
+        name: eu
+    EOF
+    ```
+
+7. In the workload clusters, create a Source for the Capact cluster:
+
+     ```yaml
+     cat <<EOF | kubectl --context kind-$CLUSTER_NAME apply -f -
+     apiVersion: multicluster.admiralty.io/v1alpha1
+     kind: Source
+     metadata:
+       name: cd
+     spec:
+       serviceAccountName: cd
+     EOF
+     ```
+
+8. Check that virtual nodes have been created in the Capact cluster to represent workload cluster:
+
+    ```bash
+    kubectl --context kind-dev-capact get nodes
+    ```
+
+9. Label the default Namespace in the Capact cluster to enable multi-cluster scheduling at the namespace level:
+
+    ```bash
+    kubectl --context kind-dev-capact label ns default multicluster-scheduler=enabled
+    ```
+
+## Demo
+
+1. Create Argo Workflow in Capact cluster, targeting the `eu` workload cluster:
+
+    ```yaml
+    cat <<EOF | kubectl --context kind-dev-capact apply -f -
+    apiVersion: batch/v1
+    kind: Job
+    metadata:
+      name: eu-foo
+      namespace: default
+    spec:
+      template:
+        metadata:
+          annotations:
+            multicluster.admiralty.io/elect: ""
+        spec:
+          nodeSelector:
+            topology.kubernetes.io/region: eu
+          containers:
+          - name: c
+            image: busybox
+            command: ["sh", "-c", "echo Processing item foo && sleep 5"]
+            resources:
+              requests:
+                cpu: 100m
+          restartPolicy: Never
+    EOF
+    ```
+
+2. Watch the Argo Workflow execution:
+
+    ```bash
+    argo watch eu-foo
+    ```
+
+3. Check Argo Workflow logs:
+
+    ```bash
+    argo log eu-foo
+    ```
+
+4. Check that the pod was scheduled on `eu` workload node:
+
+    ```bash
+    kubectl --context kind-eu get pods -o wide -n default
```

Comment:
FYI I couldn't make it able to work, but I didn't want to waste time for investigation, and also I was using k3d cluster (as the Capact cluster).

I get the idea though 🙂 

```
  ----     ------            ----   ----             -------
  Warning  FailedScheduling  7m17s  admiralty-proxy  0/2 nodes are available: 1 Insufficient cpu, 1 Too many pods, 1 node(s) didn't match node selector.
  Warning  FailedScheduling  7m17s  admiralty-proxy  0/2 nodes are available: 1 Insufficient cpu, 1 Too many pods, 1 node(s) didn't match node selector.
```

```
❯ k describe nodes admiralty-default-eu
Name:               admiralty-default-eu
Roles:              cluster
Labels:             alpha.service-controller.kubernetes.io/exclude-balancer=true
                    kubernetes.io/role=cluster
                    type=virtual-kubelet
                    virtual-kubelet.io/provider=admiralty
Annotations:        node.alpha.kubernetes.io/ttl: 0
CreationTimestamp:  Thu, 25 Nov 2021 14:33:34 +0100
Taints:             virtual-kubelet.io/provider=admiralty:NoSchedule
Unschedulable:      false
Lease:
  HolderIdentity:  admiralty-default-eu
  AcquireTime:     <unset>
  RenewTime:       Thu, 25 Nov 2021 14:43:34 +0100
Conditions:
  Type    Status  LastHeartbeatTime                 LastTransitionTime                Reason  Message
  ----    ------  -----------------                 ------------------                ------  -------
  Ready   True    Thu, 25 Nov 2021 14:43:34 +0100   Thu, 25 Nov 2021 14:33:34 +0100
Addresses:
  InternalIP:  10.42.0.124
System Info:
  Machine ID:
  System UUID:
  Boot ID:
  Kernel Version:
  OS Image:
  Operating System:
  Architecture:
  Container Runtime Version:
  Kubelet Version:
  Kube-Proxy Version:
PodCIDR:                      10.42.1.0/24
PodCIDRs:                     10.42.1.0/24
Non-terminated Pods:          (0 in total)
  Namespace                   Name    CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----    ------------  ----------  ---------------  -------------  ---
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests  Limits
  --------           --------  ------
  cpu                0 (0%)    0 (0%)
  memory             0 (0%)    0 (0%)
  ephemeral-storage  0 (0%)    0 (0%)
Events:              <none>
```